### PR TITLE
IGNITE-16165 Implement (un)marshalling of Serializable

### PR DESCRIPTION
**Pull request by rpuch**
https://issues.apache.org/jira/browse/IGNITE-16165

### CODE REVIEW

File: modules/network/src/main/java/org/apache/ignite/internal/network/serialization/marshal/StructuredObjectMarshaller.java [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+        if (layer.hasSerializationOverride()) {
+            writeWithWriteObject(object, layer, output, context);
+        } else {
+            defaultWriteFields(object, layer, output, context);
+        }
+
+        context.addUsedDescriptor(layer);
+    }
+
+
```
Comment (SammyVimes) [inline modules/network/src/main/java/org/apache/ignite/internal/network/serialization/marshal/StructuredObjectMarshaller.java:89]: Odd newline

File: modules/network/src/main/java/org/apache/ignite/internal/network/serialization/marshal/UncheckedUnmarshalException.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.network.serialization.marshal;
+
+/**
+ * An unchecked variation of {@link UnmarshalException}.
+ */
+class UncheckedUnmarshalException extends RuntimeException {
```
Comment (SammyVimes) [inline modules/network/src/main/java/org/apache/ignite/internal/network/serialization/marshal/UncheckedUnmarshalException.java:23]: Are you sure we need exceptions for both marshalling and unmarshalling?

**Review by SammyVimes — APPROVED** [review]
LGTM!

File: modules/network/src/main/java/org/apache/ignite/internal/network/serialization/marshal/StructuredObjectMarshaller.java (hunk 1 above)
Comment (rpuch) [reply modules/network/src/main/java/org/apache/ignite/internal/network/serialization/marshal/StructuredObjectMarshaller.java]: Oops, those pesky blank lines... strnge it was not caught by checkstyle. Fixed.

File: modules/network/src/main/java/org/apache/ignite/internal/network/serialization/marshal/UncheckedUnmarshalException.java (hunk 2 above)
Comment (rpuch) [reply modules/network/src/main/java/org/apache/ignite/internal/network/serialization/marshal/UncheckedUnmarshalException.java:23]: There does not seem to be a lot of harm in having separate exceptions. Advantages also seem to be minor, but it's not clear whether it's worth the effort to refactor this to only leave one exception class.

**Pull request closed without merging**